From eed8b77549e45e3791b69181086279bcf833aa1b Mon Sep 17 00:00:00 2001
From: Chintan Vankar <c-vankar@ti.com>
Date: Thu, 26 Jun 2025 11:11:24 +0530
Subject: [PATCH 47/73] PENDING: net: ethernet: ti: am65-cpsw-nuss: Update
 port_mask while adding multicast entry

Multicast entry rules are mainly evaluated for receiving traffic and do not
require MAC ports to be explicitly associated with them. However, setting
associated MAC port's bit to the port_mask of the ALE entry ensures
technical correctness and helps maintaining ALE entries linked to MAC
ports.

Signed-off-by: Chintan Vankar <c-vankar@ti.com>
---
 drivers/net/ethernet/ti/am65-cpsw-nuss.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ti/am65-cpsw-nuss.c b/drivers/net/ethernet/ti/am65-cpsw-nuss.c
index 5c2add26d4aa..bc757112a447 100644
--- a/drivers/net/ethernet/ti/am65-cpsw-nuss.c
+++ b/drivers/net/ethernet/ti/am65-cpsw-nuss.c
@@ -390,7 +390,7 @@ static void am65_cpsw_nuss_ndo_slave_set_rx_mode(struct net_device *ndev)
 	cpsw_ale_set_allmulti(common->ale,
 			      ndev->flags & IFF_ALLMULTI, port->port_id);
 
-	port_mask = ALE_PORT_HOST;
+	port_mask = BIT(port->port_id) | ALE_PORT_HOST;
 	/* Clear all mcast from ALE */
 	cpsw_ale_flush_multicast(common->ale, port_mask, -1);
 
-- 
2.51.1

